### PR TITLE
background image scaling good without jumping (2)

### LINKED ISSUE #466: Fix aspect-ratio of custom background images
custom background images are always shown as "scale-to-screen" which does not respect the aspect ration.

i would suggest to show background images always with the correct aspect ration and cut parts top/bottom or left/right (so that the image is centered in the screen rectangle).

https://github.com/deltachat/deltachat-android/issues/460#issuecomment-439395579 shows some images

there is some code in the old telegram-ui that handles this by on-the-fly-creating a temporary bitmap cropped from the orignal one iirc.

### PR DESCRIPTION
**Pull request by angelo-fuchs**
continues where #1429 left off due to rebase issues.

fixes #466

When finished needs to:

 -   Keep aspect ratio of background image in landscape and portrait orientation in the chat view
 -   Fill screen with image (no empty borders)
 -   Maintain position when the keyboard is opened.

@cyBerta Your suggestion to move the `measureViewSize` call after the early return is included.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 1]
```diff
@@ -869,16 +874,17 @@ private void initializeViews() {
...
     String backgroundImagePath = Prefs.getBackgroundImagePath(this);
+    Drawable background;
     if(!backgroundImagePath.isEmpty()) {
-      Drawable image = Drawable.createFromPath(backgroundImagePath);
-      getWindow().setBackgroundDrawable(image);
+      background = Drawable.createFromPath(backgroundImagePath);
     }
-    else if(dynamicTheme.isDarkTheme(this)) {
-      getWindow().setBackgroundDrawableResource(R.drawable.background_hd_dark);
+    else if(DynamicTheme.isDarkTheme(this)) {
```
Comment (Hocuri) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:881]: :+1: btw

File: src/org/thoughtcrime/securesms/components/ScaleStableImageView.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import static android.content.res.Configuration.ORIENTATION_PORTRAIT;
+import static android.content.res.Configuration.ORIENTATION_LANDSCAPE;
+
+/**
+ * An image view that maintains the size of the drawable provided.
+ * That means: when you set an image drawable it will be scaled to fit the screen once.
+ * If you rotate the screen it will be rescaled to fit.
+ * If you crop the screen (e.g. because the soft keyboard is displayed) the image is cropped instead.
+ *
+ * @author Angelo Fuchs
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/components/ScaleStableImageView.java:31]: in general, we do not use the @author tag, this is hard to maintain. the git history should be sufficient.

**Comment by r10s** [comment]
looking forward to try that out finally, thanks a lot @angelo-fuchs for your restless work in this issue!

**Comment by angelo-fuchs** [comment]
@r10s I tried to top align the image but did not succeed. It was then squished when the keyboard was shown.

**Comment by r10s** [comment]
>  It was then squished when the keyboard was shown.

k, i assumed sth. like that :) thanks for merging!

**Pull request merged**